db/crud.go

```diff
@@ -1754,6 +1759,11 @@ func (db *Database) updateAndReturnDoc(docid string, allowImport bool, expiry ui
 				doc.inlineSyncData = true
 			}
 
+			if existingDoc != nil {
+				doc.rawUserXattr = existingDoc.UserXattr
+			}
+			doc.rawUserXattr = currentUserXattr
```

Comment:
This looks like it's just overwriting any value that might have been in existingDoc.  What's the difference between existingDoc.UserXattr and currentUserXattr at this point?

Reply:
You're right. Not really sure why I was doing this. At this point those values would be the same. Or in fact the 'currentUserXattr' would be 'more' correct as this would be re-generated in a cas failure scenario.